app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,63 @@
+class InvitationsController < ApplicationController
+
+	def new
+		@invitation = Invitation.new
+	end
+	
+	def create
+		@invitation = Invitation.new(invitation_params)
+		@invitation.sender = current_user
+		@invitation.accepted = false
+		if @invitation.save
+			flash[:notice] = "Thank you, invitation created."
+			InvitationMailer.confirmation_email(@invitation).deliver
+			redirect_to users_path
+		else
+	    	flash[:alert] = "something is wrong :/"
+	    	redirect_to users_path
+	  end
+	end
+
+	def destroy
+		@invitation = Invitation.find(params[:id])
+
+		if @invitation.destroy
+			flash[:notice] = "Invitation deleted"
+			redirect_to users_url
+		else
+			flash[:alert] = "something is wrong :/"
+			redirect_to users_url
+	  end
+  end
+
+	def confirm_invitation
+		@invitation = Invitation.find_by_id(params[:id])
+		if @invitation
+			@user = User.new do |user|
+				user.first_name = "first_name"
+				user.last_name = "last_name"
```

Comment:
that was an idea do it that way that user is created automatically and invited person doesn't have to sign up but only change default "first_name" on his/her real name.